Adding module and state to download artifacts from Nexus Repository

anitakrueger opened the pull request:
### What does this PR do?
This PR adds a module to connect to Nexus Repository like the artifactory module does. It has been copied from the artifactory module and adapted to fit Nexus 3.0.
The accompanying state provides one state to make sure artifacts from Nexus are downloaded to a given location.
Unfortunately the artifactory module can't be used to talk to Nexus because their repository structure is different.

### What issues does this PR fix or reference?
None

### New Behavior
Artifacts can be downloaded from Nexus based on being the latest snapshot or a specific release version.

### Tests written?
No

Commit:
Initial start of ansible module

Commit:
Add initial module resolver, implement modules map

Commit:
Add help resolver on a particular Python module of Ansible.

Commit:
Improve error handling for the loaded module

Commit:
Implement module lister

Commit:
Set dynamically all possible Ansible callers as functions

Commit:
Allow long names with dots in between

Commit:
Add missed import

Commit:
Prepare arguments for the module

Commit:
Register the full name of the module

Commit:
Implement basic module caller (Python modules only)

Commit:
Catch loader error as well

Commit:
Call description from help by default

Commit:
Prevent dummy modules to be executed to nothing

Commit:
Add quotes to the help command

Commit:
Initial scaffolding for Ansible state module

Commit:
Take to the account popen's STDERR as well.

Commit:
Wrap caller into a class

Commit:
Implement basic Ansible state call

Commit:
Handle errors

Commit:
Remove dead code

Commit:
Mimic the direct call of the module to reuse internal arg parsing

Commit:
Add the license header

Commit:
Add scaffold for the unit test

Commit:
Add unit test for documentation extractor

Commit:
Do not use mock but real resolver class

Commit:
Add unit test for modules filtering

Commit:
Refactor to setup/teardown methods

Commit:
Add unit test resolver module loader failure

Commit:
Improve module loader failure

Commit:
Add unit test for module loader

Commit:
Add unit test for module loader failure

Commit:
Don't bother with non-pytest unit tests

Commit:
Lintfix: W1699

Commit:
Lintfix: E0102

Commit:
Lintfix: E9502

Commit:
Lintfix: W0108

Commit:
Lintfix: W0611

Commit:
Lintfix: E8701 (PEP8 E701), C0321

Commit:
Lintfix: W0602

Commit:
Fix TypeError for argument of type 'LoaderError' is not iterable

Commit:
Fix value is not iterable error

Commit:
Remove extra assertions

Commit:
Add more descriptive function name

Commit:
Update zfs.py

Commit:
Use timed subprocess for bailing out on hanging situation

Commit:
Add timeout option, default to 20 minutes

Commit:
Add root-level documentation to the module

anitakrueger committed:
Adding module and state to download artifacts from Nexus Repository Manager

anitakrueger commented:
I'll look at the failed checks tonight.

cachedout commented:
Hi @anitakrueger 

Glad to see you! We are moving toward a policy where all new modules/states need to have accompanying test coverage to be merged into Salt. Otherwise, they'll have to go into the `contrib` repo. Could you add tests for these? Thanks.

anitakrueger commented:
Hey Mike, yes of course. Makes sense. I need to work on those test failures anyways. Might have to wait until the weekend when I get to it.

rallytime commented:
re-run py2

rallytime commented:
Hi @anitakrueger - good to see you! I am re-running the Ubuntu tests. Those failures didn't look related to your change to me, so we'll see how this run goes. That being said, I think the only one you need to worry about right now is the `codeclimate` result.

Commit:
Change some exception imports to match the others

The `salt.exception` import was never defined, so let's change those
references to match the other defined imports.

rallytime commented on salt/modules/nexus.py:
```diff
@@ -0,0 +1,501 @@
+    try:
+        request = urllib.request.Request(artifact_url, None, headers)
+        f = urllib.request.urlopen(request)
+        with salt.utils.fopen(target_file, "wb") as local_file:
```
This has been moved to `salt.utils.files.fopen`. Can you update this reference?

anitakrueger replied:
Fixed

rallytime commented on salt/states/nexus.py:
```diff
@@ -0,0 +1,147 @@
+'''
+This state downloads artifacts from Nexus 3.x.
+
+'''
```
A version added tag would be good here as well.

anitakrueger replied:
Fixed

rallytime commented on salt/states/nexus.py:
```diff
@@ -0,0 +1,147 @@
+log = logging.getLogger(__name__)
+
+
+def downloaded(name, artifact, target_dir='/tmp', target_file=None):
```
This state should have a `__virtual__()` function to gate loading it as well.

anitakrueger replied:
Added __virtual__ function

rallytime reviewed (changes requested):
Overall, this looks great. I have a couple of minor comments that should be addressed.

mirceaulinic committed:
Improve the http_logstash engine

At the time being, this engine considers only the events
starting with salt/job. That was a poor design decision,
as the user may have other interesting events on the Salt bus,
that don't necessarily belong to the Salt standard events.
Moreover, this was not in-line with the other TCP/UDP Logstash engine
https://github.com/saltstack/salt/blob/2017.7/salt/engines/logstash.py#L76

I also improved a little the documentation.

Commit:
Add unit test for file.line on mode checks

Commit:
Rename long name variable, format a better error message

Commit:
Rename a unit test and clarify its purpose in docs

Commit:
Add unit test for file.line on an empty content while not deleting anything

Commit:
Move all file.line unit tests to the own class

Commit:
Add unit test for file.line on no location/before/after check failure

Commit:
Fix docs for skipif decorator

Commit:
Fix docs in the funtions

Commit:
Add unit test for file._starts_till function

Commit:
Update unit test data

Commit:
Add unit test for file.line insert=after

Commit:
Bugfix: file.line insert=after

Commit:
Rename unit test while using no patterns

Commit:
Add unit test for using patterns for line matching

Commit:
Remove unnecessary code

Commit:
Harden unit test for 'after pattern' scenario

Commit:
Add unit test file.line insert before line

Commit:
Fix: before/after together now define a range, where line will be added right before the last one within the range, i.e. before "before".

Commit:
Rename unit test

Commit:
Add unit test for file.line usage together with before/after

Commit:
Add unit test for file.line delete by direct match or regular expression

Commit:
Add unit test for file.line replacement

Commit:
Remove irrelevant issue number from the docstring

Commit:
Add unit test about file.line for insert/start

Commit:
Remove irrelevant issue number

Commit:
Add unit test for file.line when inserting a line at the end of the file (append functionality)

Commit:
Bugfix: content at the beginning should be added as a line, instead of merged to the rest of the data

Commit:
Bugfix: content at the end should be added as a line, instead of merged to the rest of the data

Commit:
Bugfix: keep indent of the "before" line.

Commit:
Add unit test for file.line when calling insert/ensure mode

prymitive committed:
Use pchanges in highstate output module if changes key is empty and pchanges is present

As per discussion in https://github.com/saltstack/salt/issues/40208 pchanges should be used for dry-run, it requires proper handling in the output module to correctly report changes to the user

prymitive committed:
Don't populate changes key in dry-run mode

If dry-run mode is active (test=true) only pchanges should be populated otherwise change is reported twice

Commit:
Add unit test for file.line on calling insert/ensure after specific line

garethgreenaway committed:
Merging in watch_any changes

Commit:
Add unit test for file.line ensuring line between two other lines nearby

Commit:
Add unit test for file.line ensuring line inside the range of lines

Commit:
Make assert_occurrence return amount of occurrences

Commit:
Add unit test for file.line ensuring line between two lines but content already exists

Commit:
Refactor insert/ensure when after/before is known

The-Loeki committed:
Kubernetes client certificate file usage fix

Commit:
Refactor file.line insert/ensure with after/before conditions

garethgreenaway committed:
tweaking

Commit:
Merge pull request #44273 from DSRCorporation/bugs/44239_syndic_progress

Workaround progressbar failure if minion is behind syndic.

Commit:
Fix lint violation

Commit:
Rewrite unit tests for file.line on delete/replace an empty line

benediktwerner committed:
Merge branch 'develop' into develop

Ch3LL committed:
Include file size check for test_spm_build_big_file test

benediktwerner committed:
Fixed linting

sjorge committed:
pkgin.latest_version should throw keyerror

```
KeyError: 'sabnzbd'
Traceback (most recent call last):
  File "/opt/local/bin/salt-call", line 11, in <module>
    salt_call()
  File "/opt/local/lib/python2.7/site-packages/salt/scripts.py", line 391, in salt_call
    client.run()
  File "/opt/local/lib/python2.7/site-packages/salt/cli/call.py", line 58, in run
    caller.run()
  File "/opt/local/lib/python2.7/site-packages/salt/cli/caller.py", line 134, in run
    ret = self.call()
  File "/opt/local/lib/python2.7/site-packages/salt/cli/caller.py", line 204, in call
    ret['return'] = func(*args, **kwargs)
  File "/opt/local/lib/python2.7/site-packages/salt/modules/pkgin.py", line 190, in latest_version
    return pkglist[names[0]]
KeyError: 'sabnzbd'
Traceback (most recent call last):
  File "/opt/local/bin/salt-call", line 11, in <module>
    salt_call()
  File "/opt/local/lib/python2.7/site-packages/salt/scripts.py", line 391, in salt_call
    client.run()
  File "/opt/local/lib/python2.7/site-packages/salt/cli/call.py", line 58, in run
    caller.run()
  File "/opt/local/lib/python2.7/site-packages/salt/cli/caller.py", line 134, in run
    ret = self.call()
  File "/opt/local/lib/python2.7/site-packages/salt/cli/caller.py", line 204, in call
    ret['return'] = func(*args, **kwargs)
  File "/opt/local/lib/python2.7/site-packages/salt/modules/pkgin.py", line 190, in latest_version
    return pkglist[names[0]]
KeyError: 'sabnzbd'
```

We fix this by checking if we actually have an entry for names[0] before trying to return.

sjorge committed:
Fix latest_version on newish pkgin >- 0.7.0

As per the change log in 0.7.0 the ```-p``` flag was introduced to
generate parsable output.

We assume we have this output if we have the regex support but those are
seperate. This commit adds a new internal function to detect the
support. This also fixes latest_version to now work properly on systems
with this newer version of pkgin.

sjorge committed:
Implement pkgin.list_upgrades

sjorge committed:
pkgin.upgrade does not take an array of package

```pkgin full-upgrade``` does not allow for single package upgrades.
For now we ignore the pkgs array.

sjorge committed:
Add support for `pkgs` parameters to pkgin.upgrade.

Commit:
Merge pull request #44364 from Ch3LL/fix_size_test

Include disk size check for test_spm_build_big_file test

garethgreenaway committed:
If a state is in prerequired, do not add to the self.pre to the list of requirements.

mirceaulinic committed:
Tag changes for Oxygen

vernondcole committed:
try to ping node before Wake on Lan

vernondcole committed:
Proper error message for delete of non-answering node

vernondcole committed:
soft error when no response from Vagrant ssh-config

sjorge committed:
Move zpool grains into salt.grains.zpool

sjorge committed:
Introduce a zfs_support grain

There was some ugly duplicated logic in salt.modules.zfs and
salt.modules.zpool. This got cleaned up and moved inside salt.grain.zfs.

This way the logic gets execute less often and we can simple check for
zfs_support grain in the other locations.

sjorge committed:
Update modules to use the zfs_support grain

sjorge committed:
Minor cleanup

sjorge committed:
Remove smartos specific grains from core

This allows the removal of the json module dependancy
NOTE: Not sure how the override of some unavailable smbios grains will
work when they are done in a seperate module. TIAS

sjorge committed:
Moved SmartOS grains to salt.grains.smartos

For the computenode data 3 exec's were condensed into one, execs are
expensive.

sjorge committed:
Also make available info about the type of vms

Commit:
Mute pylint on misdetected regex-like strings that are actually just test data.

Commit:
Merge branch 'develop' into smartos_grain_split

sjorge committed:
Remove _zpool_data from the bsd_memdata test for grains

sjorge committed:
Split _smartos_zone_data

This commit splits _smartos_zone_data into 3 functions,
- _smartos_zone_data => zoneid, zonename, imageversion
- _smartos_zone_pkgsrc_data => pkgsrc related data for the zone
- _smartos_zone_pkgin_data => pkgin related data for the zone

sjorge committed:
Merge branch 'smartos_grain_split' of github.com:sjorge/salt into smartos_grain_split

Commit:
Merge pull request #44347 from rallytime/merge-2017.7

[2017.7] Merge forward from 2016.11 to 2017.7

Commit:
Merge pull request #44358 from The-Loeki/kube-client-cert-file

Kubernetes client certificate file usage fix

eradman committed:
Implement swap_total grain on OpenBSD

Total swap is not reported by sysctl(8) on OpenBSD. Use swapctl(8) instead.

Signed-off-by: Eric Radman <ericshane@eradman.com>

SEJeff committed:
Add the beginning of tests for states.kubernetes and update things

* Add 100% test coverage for the kubernetes.secret_present state
* Add 100% test coverage for the kubernetes.configmap_absent state
* Add 100% test coverage for the kubernetes.configmap_present state
* Make some fields optional to kubernetes.create_secret and add examples
* Make some fields optional to kubernetes.replace_secret and add examples
* Make some fields optional to kubernetes.create_configmap and add examples
* Make some fields optional to kubernetes.replace_configmap and add examples
* Add 'comment' when a kubernetes configmap or secret is going to be replaced
* Update 'source', 'template', and 'saltenv' kwargs to defaults from the rest of the codebase
* Allow passing empty 'data' to the kubernetes.configmap_present state to remove the existing contents

anitakrueger committed:
Merge branch 'develop' of https://github.com/saltstack/salt into add-feature-nexus

anitakrueger committed:
Reducing cyclomatic complexity in nexus state and fixing a bunch of comments

anitakrueger committed:
Reducing cyclomatic complexity in nexus state and fixing a bunch of comments

anitakrueger commented on salt/modules/nexus.py:
```diff
@@ -0,0 +1,501 @@
+# -*- coding: utf-8 -*-
+'''
+Module for fetching artifacts from Nexus 3.x
+'''
```
Fixed

anitakrueger commented on salt/modules/nexus.py:
```diff
@@ -0,0 +1,501 @@
+import logging
+
+# Import Salt libs
+import salt.utils
```
Fixed

anitakrueger commented:
I'll need a couple more days to add those tests for the module and state. Bit of a learning curve for me, but I'll look at what is currently there for artifactory as this is what nexus was based on.

Commit:
Merge branch 'develop' into smartos_grain_split

SEJeff committed:
More kubernetes state tests

* Add 100% test coverage for the kubernetes.secret_absent state
* Add 100% coverage for kubernetes.node_label_present
* Also add 'comment' to kubernetes.node_add_label
* Make the faux test objects more believable in test_kubernetes.py

SEJeff committed:
Up the total test coverage of the kubernetes state to 54%!

* Add 100% test coverage for the kubernetes.node_label_absent state
* Add 100% test coverage for the kubernetes.namespace_present state
* Add 100% test coverage for the kubernetes.namespace_absent state

SEJeff committed:
Fix a bug in the kubernetes.namespace_present state

This bug found courtesy of writing unit tests to ensure it worked! When
a namespace is created via `kubernetes.namespace_present`, the 'result'
key of the return dict stays the default of `False`, which is confusing
after it is actually created.

TDD, finding your problems, just like mom!

SEJeff committed:
Fix docstring in kubernetes.namespace_present state

SEJeff committed:
Small formatting fixup

Commit:
Merge branch '2017.7' into 'develop'

Conflicts:
  - .gitignore
  - salt/modules/win_lgpo.py
  - salt/utils/__init__.py
  - tests/integration/spm/test_build.py
  - tests/unit/test_pydsl.py

Commit:
Update old utils paths to new paths

Commit:
Reduce the number of days an issue is stale by 10

Commit:
Merge pull request #44409 from rallytime/stale-config-update

Reduce the number of days an issue is stale by 10

Commit:
Merge branch 'develop' into smartos_grain_split

garethgreenaway committed:
Adding docs and integration tests.

garethgreenaway committed:
Adding state file used in integration tests.

Commit:
Fixed typo before libs import

Commit:
Fixed typo before libs import

eradman committed:
Add integration test for (mem|swap)_total grains on OpenBSD

garethgreenaway committed:
Replacing tabs with spaces.

Commit:
Merge pull request #44246 from nicholasmhughes/add-location-to-file.line-ensure

Add location support to file.line ensure

garethgreenaway committed:
Updating unit tests to limit to two required states.

prymitive committed:
Merge branch 'develop' into file-managed-test-changes

Commit:
Merge pull request #44381 from sjorge/smartos_grain_split

Split SmartOS related grain off into a seperate module

sjorge committed:
Drop zpool grains from salt.grains.core

Commit:
- fix "ValueError: too many values to unpack" bug

samodid committed:
fix docstring errors during rendering

samodid committed:
Update states docstring for better auto index description

samodid committed:
Update modules docstring for better auto index description

Commit:
file: fix file.line documentation formatting

There were mixed :param:s and simple paragraphs which resulted in a confusing
doc page.

mtkennerly committed:
Fix #44423: Handle index=None and index=0 distinctly

Commit:
Merge branch 'develop' of github.com:sadminriley/salt into develop

Conflicts:
	salt/config/schemas/common.py

sadminriley committed:
Merge branch 'develop' into develop

Commit:
Merge pull request #44380 from sjorge/zfs_grain_split

Split zfs related grain off into a seperate module

Commit:
Merge pull request #44413 from sadminriley/develop

Fixed typo before libs import on common.py

sjorge committed:
Update function comments

Commit:
Merge branch 'develop' into bugfix/issue-44423-win-path-prepend

sjorge committed:
Allow for -f to be passed to refresh_db

Commit:
Merge pull request #44421 from orymate/fix-salt-line-doc

fix states.file.line documentation formatting

bbczeuz committed:
Fix read_only doc typo

Following the examples, this should be 'rw' instead of 'ro'

garethgreenaway committed:
Updating tests for watch_any_fail and require_any_fail.

samodid committed:
fix small lint error

thatch45 committed:
make it clear that disableing the job cache is bad

Commit:
Merge pull request #44428 from bbczeuz/bbczeuz-patch-dockertypos

Fix doc typo in docker_container/read_only

Commit:
Merge pull request #44420 from samodid/fix_docstrings_rendering

Fix docstrings rendering errors

Commit:
Fix a couple of small spelling errors

Commit:
Merge pull request #44405 from SEJeff/kubernetes-updates-v2

Kubernetes updates v2

garethgreenaway committed:
fixing docs per @rallytime

bd-david-suarez committed:
Fix typo in win_lgpo module

Correct a typ in the win_lgpo module where policy was incorrectly spelled as 'polic'

Commit:
Merge pull request #44430 from thatch45/doc_fix1

make it clear that disabling the job cache is bad

Commit:
Merge pull request #44353 from prymitive/file-managed-test-changes

File managed test changes

mtkennerly committed:
Merge branch 'develop' into bugfix/issue-44423-win-path-prepend

Commit:
Merge pull request #44433 from dave92082/fix-typo-in-lgpo-module

Fix typo in win_lgpo module

Commit:
Merge pull request #44424 from mtkennerly/bugfix/issue-44423-win-path-prepend

Fix #44423: Handle index=None and index=0 distinctly in the win_path.exists state

terminalmage committed:
Use with blocks (or try/finally) to ensure filehandles are closed

This fixes a few sources (or potential sources) of filehandle leaks.

Commit:
Add a late-import for the salt.utils.master.py file in salt.auth

Fixes #44435

PR #43703 inadvertently moved a "late import" statement up to the
top of the auth file, which caused the problem reported in the PR
itself and issue #44435.

Moving the salt.utils.master import lower down in the file fixes
the issue.

Commit:
Merge pull request #44407 from rallytime/merge-develop

[develop] Merge forward from 2017.7 to develop

Commit:
Merge pull request #44403 from eradman/swap_total

Implement swap_total grain on OpenBSD

Commit:
Merge pull request #44415 from vernondcole/vagrant_cloud_minor_revisions

Vagrant and saltify cloud minor revisions

sjorge committed:
Simpler fallback check for zfs-fuse.

This should fix #44447

connordelacruz committed:
Fixed a bug with change output when not pruning

Union operator was being used on a set and a list. Cast the list to a
set, then cast the resulting set from the union back into a list before
adding it to ret['changes']

terminalmage committed:
Fix stacklevel in salt/utils/__init__.py

terminalmage committed:
Wait until after restoring original resource limit before removing temp files

Invoking shutil.rmtree() first was causing a traceback since the max
open files setting was still set very low for the test. This moves the
shutil.rmtree() until after we have restored the original resource
limit.

garethgreenaway committed:
Updating documentation and release notes.

samodid committed:
fix #38838

  update aptpkg.list_pkgs behavior

samodid committed:
fix #38838

  update aptpkg.list_pkgs behavior

terminalmage committed:
Python 3 compatibility fixes for kubernetes

dict.keys() is not a list in Python 3, and base64.b64encode requires
a bytestring.

terminalmage committed:
Update reference to old salt.utils.which function

DmitryKuzmenko committed:
Do not do the same request twice.

Found during code review. Originally there was 2 different functions to
check is user there and to fetch the users list. Then the first one was
replaced by more efficient that is the same that the second one.

Removed the unneeded second call.

samodid committed:
move decision statement to upper if block

mephi42 committed:
nova: fix endpoint URL determination in _v3_setup()

Some OpenStack setups return multiple entries for the same service.
This change makes the code consider all matching entries instead of
just the first one.

Commit:
Merge pull request #44442 from terminalmage/fix-leaked-filehandles

Use with blocks (or try/finally) to ensure filehandles are closed

Commit:
Merge branch 'develop' into fix_38838

Commit:
Discard Git's spaghetti merger

Commit:
Merge pull request #44445 from rallytime/fix-44435

Add a late-import for the salt.utils.master.py file in salt.auth

Commit:
Merge pull request #44468 from DSRCorporation/bugs/mongo_users_do_not_fetch_twice

Do not perform the same request twice.

Commit:
Merge pull request #44448 from sjorge/zfsfuse

Simpler fallback check for zfs-fuse.

Commit:
Merge branch 'develop' into develop

Commit:
Merge branch 'develop' into pkgin_uptodate

Commit:
Reduce the number of days an issue is stale by 10

gtmanfred committed:
update kitchen yaml to use master for salt-jenkins

Commit:
Merge pull request #44481 from rallytime/stale-config-update

Reduce the number of days an issue is stale by 10

vitaliyf committed:
Fixed minor spelling mistake.

Commit:
Merge branch 'develop' into separate-key-dir-from-cache-dir

Commit:
Merge branch 'develop' into pkgin_uptodate

Commit:
Merge pull request #44356 from skjaro/ipset-fix-subtypes-comma

Fix ipset state with multiple entries and subtypes separated with comma

Commit:
Merge branch 'develop' into develop

Commit:
Merge pull request #44309 from cloudflare/http-logstash-eng

Improve the http_logstash engine

Commit:
Merge pull request #44288 from eradman/onlyif_unless

Report 'onlyif' and 'unless' conditions as true or false

Commit:
Remove unused variable

Commit:
Use tuple instead of list for joining

cachedout commented:
@anitakrueger Sounds good. Keep us posted. Thanks!

Commit:
Merge pull request #44148 from roaldnefs/add-telegram-returner

Add Telegram returner

Commit:
Merge pull request #44486 from vitaliyf/patch-1

Fixed minor spelling mistake.

Commit:
Merge pull request #44482 from gtmanfred/develop

update kitchen yaml to use master for salt-jenkins

Commit:
Merge pull request #44361 from isbm/isbm-file.line-issues-devel

Long-awaiting set of unit tests and fixes for "file.line"

sjorge committed:
Add the missing versionchanged info.

anitakrueger committed:
Making classifier optional in get_snapshot_version_string

anitakrueger committed:
Making classifier optional in get_snapshot_version_string

anitakrueger committed:
Adding unit tests for nexus state and module based on artifactory tests

anitakrueger committed:
Adding unit tests for nexus state and module based on artifactory tests

anitakrueger commented:
Added the tests and ran them using
```
python tests/runtests.py -n unit.states.test_nexus
```
and
```
python tests/runtests.py -n unit.modules.test_nexus
```
All tests were successful.

rallytime commented:
@anitakrueger Awesome! Those new tests are running on our Jenkins runs too. 👌 

The last thing to clean up here before a final review are a couple of pylint warnings: https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/16315/violations/file/salt/states/nexus.py/

Do you have a moment to clean those up? After that, we can update the branch and that should get the Py3 tests happy, too. (They were broken downstream for a bit, but we've fixed them now.)

Commit:
Merge pull request #44502 from sjorge/PR43996

Additional comments required for PR #43996

Commit:
Add priority field to support the latest capirca.

Commit:
Add priority field to support the latest capirca.

garethgreenaway committed:
Merge branch 'develop' into req_any

sjorge committed:
Add versionadded where needed

sjorge committed:
Merge branch 'pkgin_uptodate' of github.com:sjorge/salt into pkgin_uptodate

Commit:
Merge pull request #44276 from rallytime/salt-cli-imports

Change some exception imports to match the others

Commit:
Merge pull request #44472 from mephi42/develop

nova: fix endpoint URL determination in _v3_setup()

Commit:
Merge pull request #44455 from samodid/fix_38838

Fix for #38838

Commit:
Merge pull request #44419 from rongzeng54/develop

- fix "ValueError: too many values to unpack" bug

Commit:
Merge pull request #44281 from ze42/compound_command

minion: don't hang on compound-command

Commit:
Merge pull request #44196 from benediktwerner/develop

Fix autosign file permission check [develop]

Commit:
Merge pull request #43822 from comsul/issues/42512

check_result: Correctly check the __extend__ state.

Commit:
Merge pull request #44508 from mzbroch/capirca_integration

Capirca integration

Commit:
Merge pull request #44414 from garethgreenaway/req_any

[develop] Adding additional requisites to state compiler

Commit:
Merge pull request #43780 from connordelacruz/develop

Fix issues #38855, #41075, #41717, and #43912 with firewalld state and execution modules

Commit:
Merge pull request #43504 from isbm/isbm-ansible-module-dev

Got Ansible own modules? Run them from within the Salt just like that!

Commit:
Merge pull request #43406 from rtx3/consul-module-add-acl

Add ACL token to Consul module

Commit:
Merge pull request #43474 from kstreee/separate-key-dir-from-cache-dir

Separates 'key_dir' from 'cache_dir', The key files (i.e. '.root_key', '.sudo_ACCOUNT_NAME') must not be shared with other masters.

Commit:
Merge pull request #43410 from skob/develop

queue.insert_runner with sqlite backend

Commit:
Merge pull request #42930 from CorvinM/issue36942

Jinja line statements and comments

Commit:
Merge pull request #44369 from sjorge/pkgin_uptodate

Implement pkg.uptodate for pkgin

anitakrueger committed:
Merge branch 'add-feature-nexus' of https://github.com/anitakrueger/salt into add-feature-nexus

anitakrueger committed:
Fixing pylint errors in new nexus state

anitakrueger commented:
@rallytime Sorry, it looks like I made a mistake when rebasing develop onto my new branch :( I did fix the pylint errors. Argh. How do I fix this please?

anitakrueger commented:
Looks like I need a lesson in github branching :( I did a pull upstream on my develop branch and then a rebase onto the add-feature-nexus branch. I am sure I did this before and it didn't look like this.

rallytime commented:
Hmm...I am not sure offhand what happened here, but it looks like maybe you rebased develop against your fork instead of saltstack's. I usually do something like this (assuming the saltstack repo is `upstream`, and not your github fork):
```
git fetch upstream
git checkout develop
git reset --hard upstream/develop
git checkout add-feature-nexus
git rebase develop
```
Does that help?

anitakrueger commented:
@rallytime something must have gone wrong with my branch :(
even the clean rebase produces a conflict now.
I'm opening a new clean PR. Sorry for all this mess.

anitakrueger commented:
Closed in favor of https://github.com/saltstack/salt/pull/44591

rallytime commented:
@anitakrueger No problem at all! Thanks for the addition. :)
